src/s2/util/math/exactfloat/bignum_test.cc

```diff
@@ -0,0 +1,1220 @@
+  BN_CTX* ctx = BN_CTX_new();
+  for (const auto& number : GetParam()) {
+    // Test same number multiplication (most likely to trigger edge cases)
+    const Bignum bn_a = *Bignum::FromString(number);
+    const Bignum bn_result = Bignum(2) * bn_a;
+
+    const OpenSSLBignum ssl_a(number);
+    OpenSSLBignum ssl_result;
+    BN_mul(ssl_result.get(), OpenSSLBignum("2").get(), ssl_a.get(), ctx);
+
+    // Compare string representations
+    char* ssl_str = BN_bn2dec(ssl_result.get());
+    std::string bn_str = absl::StrFormat("%v", bn_result);
+
+    EXPECT_EQ(bn_str, std::string(ssl_str))
+        << "Mismatch for multiplication"
+        << "\nBignum result: " << bn_str.substr(0, 100) << "..."
+        << "\nOpenSSL result: " << std::string(ssl_str).substr(0, 100) << "...";
+    OPENSSL_free(ssl_str);
+  }
+  BN_CTX_free(ctx);
+}
+
+TEST_P(VsOpenSSLTest, AdditionCorrect) {
+  // Test that addition produces correct results by comparing to OpenSSL.
+  const std::vector<std::string> numbers = GetParam();
+  for (size_t i = 0; i < numbers.size(); ++i) {
+    const auto& num_a = numbers[i];
+    const auto& num_b = numbers[(i + 1) % numbers.size()];
+
+    const Bignum bn_a = *Bignum::FromString(num_a);
+    const Bignum bn_b = *Bignum::FromString(num_b);
+
+    const Bignum bn_result = bn_a + bn_b;
+
+    const OpenSSLBignum ssl_a(num_a);
+    const OpenSSLBignum ssl_b(num_b);
+    OpenSSLBignum ssl_result;
+    BN_add(ssl_result.get(), ssl_a.get(), ssl_b.get());
+
+    // Compare string representations
+    char* ssl_str = BN_bn2dec(ssl_result.get());
+    std::string bn_str = absl::StrFormat("%v", bn_result);
+
+    EXPECT_EQ(bn_str, std::string(ssl_str))
+        << "Mismatch for addition"
+        << "\nBignum result: " << bn_str.substr(0, 100) << "..."
+        << "\nOpenSSL result: " << std::string(ssl_str).substr(0, 100) << "...";
+    OPENSSL_free(ssl_str);
+  }
+}
+
+TEST_P(VsOpenSSLTest, SubtractionCorrect) {
+  // Test that subtraction produces correct results by comparing to OpenSSL.
+  const std::vector<std::string> numbers = GetParam();
+  for (size_t i = 0; i < numbers.size(); ++i) {
+    const auto& num_a = numbers[i];
+    const auto& num_b = numbers[(i + 1) % numbers.size()];
+
+    const Bignum bn_a = *Bignum::FromString(num_a);
+    const Bignum bn_b = *Bignum::FromString(num_b);
+
+    const Bignum bn_result = bn_a - bn_b;
+
+    const OpenSSLBignum ssl_a(num_a);
+    const OpenSSLBignum ssl_b(num_b);
+    OpenSSLBignum ssl_result;
+    BN_sub(ssl_result.get(), ssl_a.get(), ssl_b.get());
+
+    // Compare string representations
+    char* ssl_str = BN_bn2dec(ssl_result.get());
+    std::string bn_str = absl::StrFormat("%v", bn_result);
+
+    EXPECT_EQ(bn_str, std::string(ssl_str))
+        << "Mismatch for addition"
+        << "\nBignum result: " << bn_str.substr(0, 100) << "..."
+        << "\nOpenSSL result: " << std::string(ssl_str).substr(0, 100) << "...";
+    OPENSSL_free(ssl_str);
+  }
+}
+
+absl::BitGen bitgen;
```

Comment:
This static variable is not great. Maybe the bitgen should be in the class and the params can be the sizes.